detection-rules/open_redirect_asemailmgmteu.yml

```
@@ -0,0 +1,38 @@
+name: "Open Redirect: asemailmgmteu.com"
+description: "Message contains use of the asemailmgmteu.com open redirect. This has been exploited in the wild."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.root_domain == "asemailmgmteu.com"
+          and strings.icontains(.href_url.query_params, 'redirect=')
+          // negate use of the redirect by asemailmgmteu.com
+          and not regex.icontains(.href_url.query_params,
+                                  'redirect=[^&]*asemailmgmteu\.com'
+          )
+  )
+  // negate legitimate use of the redirect in newsletters
+  and sender.email.domain.root_domain not in~ (
+      "wizznews.com",
+      "tomtom.com"
+  )
+
```

we can make use of the stirngs.parse_url and query_params_decoded to better inspect the redirect element, and limit the total body links as well. 

https://platform.sublime.security/messages/hunt?huntId=0198f925-88fb-773a-b758-34bceae732a6
hunt looks great

```suggestion
  type.inbound
  and 0 < length(body.links) < 10
  and any(body.links,
          .href_url.domain.root_domain == "asemailmgmteu.com"
          and strings.icontains(.href_url.query_params, 'redirect=')
          // negate use of the redirect by asemailmgmteu.com
          and not any(.href_url.query_params_decoded["redirect"],
                      strings.parse_url(.).domain.root_domain == sender.email.domain.root_domain
                      and not strings.parse_url(.).domain.root_domain == "asemailmgmteu.com"
          )
  )
```

---

looks great!